src/plugin/ogc/wfs/wfslayerconfig.js

```diff
@@ -352,94 +357,113 @@ plugin.ogc.wfs.WFSLayerConfig.prototype.addMappings = function(layer, options) {
 
 
 /**
- * @type {Array<RegExp>}
+ * The available WFS type config objects. Plugins can add supported parser types to this.
+ * @type {Array<os.ogc.WFSTypeConfig>}
  */
-plugin.ogc.wfs.WFSLayerConfig.PREFERRED_TYPES = [
-  /^application\/json$/,
-  /gml\/?3/i,
-  /gml\/?2/i
+plugin.ogc.wfs.WFSLayerConfig.TYPE_CONFIGS = [
+  {
+    regex: /^application\/json$/,
+    parser: 'geojson',
+    type: 'geojson',
+    priority: 100
+  },
+  {
+    regex: /gml\/?3/i,
+    parser: 'gml',
+    type: 'gml3',
+    priority: 50
+  },
+  {
+    regex: /gml\/?2/i,
+    parser: 'gml',
+    type: 'gml2',
+    priority: -100
+  }
 ];
 
 
 /**
- * @enum {number}
+ * Register a type config object.
+ * @param {!os.ogc.WFSTypeConfig} config
+ * @protected
  */
-plugin.ogc.wfs.WFSLayerConfig.TYPES = {
-  GEOJSON: 0,
-  GML3: 1,
-  GML2: 2
+plugin.ogc.wfs.WFSLayerConfig.registerType = function(config) {
+  const configs = plugin.ogc.wfs.WFSLayerConfig.TYPE_CONFIGS;
+
+  if (!configs.find((c) => c.type === config.type)) {
+    configs.push(config);
+  }
+
+  // sort them by priority, highest first
+  configs.sort((a, b) => goog.array.defaultCompare(b.priority, a.priority));
```

Comment:
You can simplify numeric sorts (and avoid using `defaultCompare`) with:
`(a, b) => b.priority - a.priority`